File: 1935-maximum-number-of-words-you-can-type/1935.py
class Solution:
    def canBeTypedWords(self, text: str, brokenLetters: str) -> int:
        count = 0
        table = set(brokenLetters)

        words = text.split(' ')
        for word in words:
            for w in word:
                if w in table:
                    count += 1
                    break

        return len(words) - count

File: 1935-maximum-number-of-words-you-can-type/1935.cpp
class Solution {
private:
    vector<string> split(const string text) {
        vector<string> res;
        istringstream iss(text);
        string word;
        while (iss >> word)
            res.push_back(word);
        return res;
    }
public:
    int canBeTypedWords(string text, string brokenLetters) {
        vector<string> words = split(text);
        int count = 0;
        set<char> s;
        for (const char broken : brokenLetters) {
            s.insert(broken);
        }
        for (const string& word : words) {
            bool b = true;
            for (const char ch : word) {
                if (s.find(ch) != s.end()) {
                    b = false;
                    break;
                }
            }
            if (b)
                ++count;
        }
        return count;
    }
};
